pkg/cloudprovider/providers/azure/azure_managedDiskController.go

```diff
@@ -201,3 +259,58 @@ func getResourceGroupFromDiskURI(diskURI string) (string, error) {
 	}
 	return fields[4], nil
 }
+
+// GetLabelsForVolume implements PVLabeler.GetLabelsForVolume
+func (c *Cloud) GetLabelsForVolume(ctx context.Context, pv *v1.PersistentVolume) (map[string]string, error) {
+	// Ignore if not AzureDisk.
+	if pv.Spec.AzureDisk == nil {
+		return nil, nil
+	}
+
+	// Ignore any volumes that are being provisioned
+	if pv.Spec.AzureDisk.DiskName == volume.ProvisionedVolumeName {
+		return nil, nil
+	}
+
+	return c.GetAzureDiskLabels(pv.Spec.AzureDisk.DataDiskURI)
+}
+
+// GetAzureDiskLabels gets availability zone labels for Azuredisk.
+func (c *Cloud) GetAzureDiskLabels(diskURI string) (map[string]string, error) {
```

Comment:
This function should be part of `DiskClient` interface and should not be part of exported Cloud implementation

Reply:
It should because it's also used in admission controller.